File: app/src/main/java/com/jintao/vipmanager/database/bean/DbUserConsumeInfo.java
package com.jintao.vipmanager.database.bean;

import org.greenrobot.greendao.annotation.Entity;
import org.greenrobot.greendao.annotation.Id;
import org.greenrobot.greendao.annotation.Property;
import org.greenrobot.greendao.annotation.Generated;

/**
 * Author: jintao
 * CreateDate: 2023/8/23 17:23
 * Description:
 */
@Entity(nameInDb = "consume_record_info")
public class DbUserConsumeInfo {

    @Id(autoincrement = true)
    private Long id;

    @Property(nameInDb = "uid")
    private int uid = 0;

    @Property(nameInDb = "content")
    private String content = "";

    @Property(nameInDb = "image_path")
    private String imagePath = "";

    @Property(nameInDb = "consume_time")
    private String consumeTime = "";

    @Property(nameInDb = "consume_amount")
    private float consumeAmount = 0;

    @Property(nameInDb = "consume_integral")
    private float consumeIntegral = 0;

    @Property(nameInDb = "conver_ratio")
    private String converRatio = "1";

    @Property(nameInDb = "consume_type")
    private boolean consumeType = false;


    @Generated(hash = 480178616)
    public DbUserConsumeInfo(Long id, int uid, String content, String imagePath,
            String consumeTime, float consumeAmount, float consumeIntegral,
            String converRatio, boolean consumeType) {
        this.id = id;
        this.uid = uid;
        this.content = content;
        this.imagePath = imagePath;
        this.consumeTime = consumeTime;
        this.consumeAmount = consumeAmount;
        this.consumeIntegral = consumeIntegral;
        this.converRatio = converRatio;
        this.consumeType = consumeType;
    }

    @Generated(hash = 547378404)
    public DbUserConsumeInfo() {
    }


    public Long getId() {
        return this.id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getImagePath() {
        return this.imagePath;
    }

    public void setImagePath(String imagePath) {
        this.imagePath = imagePath;
    }

    public String getConsumeTime() {
        return this.consumeTime;
    }

    public void setConsumeTime(String consumeTime) {
        this.consumeTime = consumeTime;
    }

    public float getConsumeIntegral() {
        return this.consumeIntegral;
    }

    public void setConsumeIntegral(float consumeIntegral) {
        this.consumeIntegral = consumeIntegral;
    }

    public boolean getConsumeType() {
        return this.consumeType;
    }

    public void setConsumeType(boolean consumeType) {
        this.consumeType = consumeType;
    }

    public int getUid() {
        return this.uid;
    }

    public void setUid(int uid) {
        this.uid = uid;
    }

    public String getContent() {
        return this.content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public float getConsumeAmount() {
        return consumeAmount;
    }

    public void setConsumeAmount(float consumeAmount) {
        this.consumeAmount = consumeAmount;
    }

    public boolean isConsumeType() {
        return consumeType;
    }

    public String getConverRatio() {
        return converRatio;
    }

    public void setConverRatio(String converRatio) {
        this.converRatio = converRatio;
    }
}

File: app/src/main/java/com/jintao/vipmanager/database/bean/DbUserConsumeInfo.kt
package com.jintao.vipmanager.database.bean

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

/**
 * Author: jintao
 * CreateDate: 2023/8/23 17:23
 * Description:
 */
@Entity(tableName = "consume_record_info")
class DbUserConsumeInfo {

    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "_id")
    private var id:Long? = 0L

    @ColumnInfo(name = "uid")
    private var uid:Int = 0

    @ColumnInfo(name = "content")
    private var content:String? = ""

    @ColumnInfo(name = "image_path")
    private var imagePath:String? = ""

    @ColumnInfo(name = "consume_time")
    private var consumeTime:String? = ""

    @ColumnInfo(name = "consume_amount")
    private var consumeAmount:Float = 0f

    @ColumnInfo(name = "consume_integral")
    private var consumeIntegral:Float = 0f

    @ColumnInfo(name = "conver_ratio")
    private var converRatio:String? = "1"

    @ColumnInfo(name = "consume_type")
    private var consumeType:Int = 0

    fun getId():Long {
        return id!!
    }

    fun setId(id:Long) {
        this.id = id
    }

    fun getUid():Int {
        return uid
    }

    fun setUid(uid:Int) {
        this.uid = uid
    }

    fun getImagePath():String {
        if(this.imagePath==null) {
            return ""
        }else {
            return this.imagePath!!
        }
    }

    fun setImagePath(imagePath: String) {
        this.imagePath = imagePath;
    }

    fun getConsumeTime():String {
        if(this.consumeTime==null) {
            return ""
        }else {
            return this.consumeTime!!
        }
    }

    fun setConsumeTime(consumeTime: String) {
        this.consumeTime = consumeTime
    }

    fun getConsumeIntegral():Float {
        return this.consumeIntegral
    }

    fun setConsumeIntegral(consumeIntegral:Float) {
        this.consumeIntegral = consumeIntegral
    }

    fun getConsumeType():Int {
        return this.consumeType
    }

    fun setConsumeType(consumeType:Int) {
        this.consumeType = consumeType
    }

    fun getContent(): String {
        if(this.content==null) {
            return ""
        }else {
            return this.content!!
        }
    }

    fun setContent(content: String) {
        this.content = content
    }

    fun getConsumeAmount():Float {
        return consumeAmount
    }

    fun setConsumeAmount(consumeAmount:Float) {
        this.consumeAmount = consumeAmount
    }

    fun getConverRatio(): String {
        if(this.converRatio==null) {
            return ""
        }else {
            return this.converRatio!!
        }
    }

    fun setConverRatio(converRatio: String) {
        this.converRatio = converRatio
    }
}
